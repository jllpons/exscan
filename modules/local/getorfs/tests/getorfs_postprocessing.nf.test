nextflow_process {

    name "Test Process GETORFS_POSTPROCESSING"
    script "../main.nf"
    process "GETORFS_POSTPROCESSING"

    test("Testing process GETORFS_POSTPROCESSING on *E. coli* LexA GETORFS output...") {


        when {
            params {
                outdir = "tests/results"
            }
            process {
                """
                input[0] = file("${projectDir}/data/fasta/lexA_Ecoli.orfs.fasta")
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }
    }
}
