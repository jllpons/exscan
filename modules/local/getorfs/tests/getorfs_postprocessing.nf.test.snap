{
    "Testing process GETORFS_POSTPROCESSING on *E. coli* LexA GETORFS output...": {
        "content": [
            {
                "0": [
                    "lexA_Ecoli.orfs.pp.fasta:md5,9832e86e3eab2be8b7b160dac5ad922b"
                ],
                "1": [
                    "versions.yml:md5,ad58e2b936279fb8132ef5f6a635a57e"
                ],
                "orfs_pp": [
                    "lexA_Ecoli.orfs.pp.fasta:md5,9832e86e3eab2be8b7b160dac5ad922b"
                ],
                "versions": [
                    "versions.yml:md5,ad58e2b936279fb8132ef5f6a635a57e"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-10-28T12:10:11.442346"
    }
}